ra/registration-authority.go

```diff
@@ -166,6 +166,12 @@ func (ra *RegistrationAuthorityImpl) NewCertificate(req core.CertificateRequest,
 	logEvent.CommonName = csr.Subject.CommonName
 	logEvent.Names = csr.DNSNames
 
+	if len(csr.DNSNames) == 0 {
```

Comment:
This is incorrect.  There could be a name in csr.Subject.CommonName.  But actually, please go ahead and drop this.  I've made some revisions to this area today in #262, so I'll go ahead and fold this change into that PR.